In [1]:
import pandas as pd
import json

dataf = pd.read_json('./Data/tweets1.json')

json_file1= json.loads(open('./Data/tweets1.json').read())

#from pprint import pprint

with open('./Data/tweets1.json') as data_file:  

    data = json.load(data_file)

#pprint(data["Blog"])
#pprint(data["Posts"]["PostX"])
#pprint(data["Tags"][1])

In [2]:
prueba1 = {"Fruteria": [  {"Fruta":   [    {"Nombre":"Manzana","Cantidad":10},    {"Nombre":"Pera","Cantidad":20},    {"Nombre":"Naranja","Cantidad":30}   ]  },  {"Verdura":   [    {"Nombre":"Lechuga","Cantidad":80},    {"Nombre":"Tomate","Cantidad":15},    {"Nombre":"Pepino","Cantidad":50}   ]  } ]}

In [3]:
print prueba1
print type(prueba1)
print repr(prueba1)
print type(repr(prueba1))

{'Fruteria': [{'Fruta': [{'Nombre': 'Manzana', 'Cantidad': 10}, {'Nombre': 'Pera', 'Cantidad': 20}, {'Nombre': 'Naranja', 'Cantidad': 30}]}, {'Verdura': [{'Nombre': 'Lechuga', 'Cantidad': 80}, {'Nombre': 'Tomate', 'Cantidad': 15}, {'Nombre': 'Pepino', 'Cantidad': 50}]}]}
<type 'dict'>
{'Fruteria': [{'Fruta': [{'Nombre': 'Manzana', 'Cantidad': 10}, {'Nombre': 'Pera', 'Cantidad': 20}, {'Nombre': 'Naranja', 'Cantidad': 30}]}, {'Verdura': [{'Nombre': 'Lechuga', 'Cantidad': 80}, {'Nombre': 'Tomate', 'Cantidad': 15}, {'Nombre': 'Pepino', 'Cantidad': 50}]}]}
<type 'str'>


In [4]:
prueba1_string = json.dumps(prueba1)
print prueba1_string

print type(prueba1_string)

{"Fruteria": [{"Fruta": [{"Nombre": "Manzana", "Cantidad": 10}, {"Nombre": "Pera", "Cantidad": 20}, {"Nombre": "Naranja", "Cantidad": 30}]}, {"Verdura": [{"Nombre": "Lechuga", "Cantidad": 80}, {"Nombre": "Tomate", "Cantidad": 15}, {"Nombre": "Pepino", "Cantidad": 50}]}]}
<type 'str'>


In [5]:
json.loads(prueba1_string)

{u'Fruteria': [{u'Fruta': [{u'Cantidad': 10, u'Nombre': u'Manzana'},
    {u'Cantidad': 20, u'Nombre': u'Pera'},
    {u'Cantidad': 30, u'Nombre': u'Naranja'}]},
  {u'Verdura': [{u'Cantidad': 80, u'Nombre': u'Lechuga'},
    {u'Cantidad': 15, u'Nombre': u'Tomate'},
    {u'Cantidad': 50, u'Nombre': u'Pepino'}]}]}

In [6]:
df_prueba = pd.read_json(prueba1_string)
df_prueba

Fruteria
0  {u'Fruta': [{u'Nombre': u'Manzana', u'Cantidad...
1  {u'Verdura': [{u'Nombre': u'Lechuga', u'Cantid...

In [7]:
data = open('./Data/tweets1.json')
print type(data)
print type(data.next())
data.seek(0)
print data.next()

<type 'file'>
<type 'str'>
{"_id":"838720064958443520","truncated":false,"text":"Excelente semana de labores a tod@s (@ Instituto Antonio Peña Celi in Loja) https://t.co/BX4CmeSPvl","is_quote_status":false,"id":{"$numberLong":"838720064958443520"},"favorite_count":0,"textFix":"excelente semana de labores a tod@s (@ instituto antonio pena celi in loja) https://t.co/bx4cmespvl","sourceFix":"\u003ca href=\"http://foursquare.com\" rel=\"nofollow\"\u003efoursquare\u003c/a\u003e","created_at_localtime":{"$date":"2017-03-06T06:57:00.000Z"},"retweeted":false,"coordinates":{"type":"Point","coordinates":[-79.20304415,-3.99452971]},"timestamp_ms":"1488801420808","entities":{"user_mentions":[],"symbols":[],"hashtags":[],"urls":[{"url":"https://t.co/BX4CmeSPvl","indices":[76,99],"expanded_url":"https://www.swarmapp.com/c/ejHEvxEoYJs","display_url":"swarmapp.com/c/ejHEvxEoYJs"}]},"retweet_count":0,"favorited":false,"user":{"profile_use_background_image":true,"default_profile_image":false,"id":525242

In [8]:
import requests
r = requests.get('http://api.open-notify.org/iss-now.json')
print r.content
print type(r)
print type(r.content)

{"iss_position": {"latitude": "32.2193", "longitude": "14.7556"}, "message": "success", "timestamp": 1492706687}
<class 'requests.models.Response'>
<type 'str'>


In [9]:
pos = json.loads(r.content)
print pos
print type(pos)

{u'timestamp': 1492706687, u'message': u'success', u'iss_position': {u'latitude': u'32.2193', u'longitude': u'14.7556'}}
<type 'dict'>


In [10]:
data.seek(0)
linea1 = pd.read_json(data.next())
print type(linea1)
print linea1.columns
print linea1.shape
linea1.head(5)

<class 'pandas.core.frame.DataFrame'>
Index([u'_id', u'coordinates', u'created_at', u'created_at_localtime',
       u'entities', u'favorite_count', u'favorited', u'filter_level', u'geo',
       u'id', u'is_quote_status', u'lang', u'place', u'possibly_sensitive',
       u'retweet_count', u'retweeted', u'source', u'sourceFix', u'text',
       u'textFix', u'timestamp_ms', u'timezone_info', u'truncated', u'user'],
      dtype='object')
(58, 24)


_id coordinates          created_at  \
$date         838720064958443520         NaN 2017-03-06 11:57:00   
$numberLong   838720064958443520         NaN                 NaT   
DST           838720064958443520         NaN                 NaT   
attributes    838720064958443520         NaN                 NaT   
bounding_box  838720064958443520         NaN                 NaT   

                  created_at_localtime entities  favorite_count favorited  \
$date         2017-03-06T06:57:00.000Z      NaN               0     False   
$numberLong                        NaN      NaN               0     False   
DST                                NaN      NaN               0     False   
attributes                         NaN      NaN               0     False   
bounding_box                       NaN      NaN               0     False   

             filter_level  geo            id ...  retweet_count retweeted  \
$date                 low  NaN           NaN ...              0     False   
$numberLong           low  NaN  8.387201e+17 ...              0     False   
DST                   low  NaN           NaN ...              0     False   
attributes            low  NaN           NaN ...              0     False   
bounding_box          low  NaN           NaN ...              0     False   

                                                         source  \
$date         <a href="http://foursquare.com" rel="nofollow"...   
$numberLong   <a href="http://foursquare.com" rel="nofollow"...   
DST           <a href="http://foursquare.com" rel="nofollow"...   
attributes    <a href="http://foursquare.com" rel="nofollow"...   
bounding_box  <a href="http://foursquare.com" rel="nofollow"...   

                                                      sourceFix  \
$date         <a href="http://foursquare.com" rel="nofollow"...   
$numberLong   <a href="http://foursquare.com" rel="nofollow"...   
DST           <a href="http://foursquare.com" rel="nofollow"...   
attributes    <a href="http://foursquare.com" rel="nofollow"...   
bounding_box  <a href="http://foursquare.com" rel="nofollow"...   

                                                           text  \
$date         Excelente semana de labores a tod@s (@ Institu...   
$numberLong   Excelente semana de labores a tod@s (@ Institu...   
DST           Excelente semana de labores a tod@s (@ Institu...   
attributes    Excelente semana de labores a tod@s (@ Institu...   
bounding_box  Excelente semana de labores a tod@s (@ Institu...   

                                                        textFix  \
$date         excelente semana de labores a tod@s (@ institu...   
$numberLong   excelente semana de labores a tod@s (@ institu...   
DST           excelente semana de labores a tod@s (@ institu...   
attributes    excelente semana de labores a tod@s (@ institu...   
bounding_box  excelente semana de labores a tod@s (@ institu...   

                        timestamp_ms timezone_info truncated user  
$date        2017-03-06 11:57:00.808           NaN     False  NaN  
$numberLong  2017-03-06 11:57:00.808           NaN     False  NaN  
DST          2017-03-06 11:57:00.808         False     False  NaN  
attributes   2017-03-06 11:57:00.808           NaN     False  NaN  
bounding_box 2017-03-06 11:57:00.808           NaN     False  NaN  

[5 rows x 24 columns]

In [11]:
test='''{"a":1,"b":2}
{"a":3,"b":4}'''
#df=pd.read_json(test,orient='records') doesn't work, expects []

l=[ json.loads(l) for l in test.splitlines()]
df=pd.DataFrame(l)
df

a  b
0  1  2
1  3  4

In [12]:
data.seek(0)
l=[ json.loads(l) for l in data]
df=pd.DataFrame(l)
df.head()

_id                                        coordinates  \
0  838720064958443520  {u'type': u'Point', u'coordinates': [-79.20304...   
1  838720176581447681  {u'type': u'Point', u'coordinates': [-79.5333,...   
2  838720534225510404  {u'type': u'Point', u'coordinates': [-78.5833,...   
3  838720814002339840  {u'type': u'Point', u'coordinates': [-78.73340...   
4  838720828053270528  {u'type': u'Point', u'coordinates': [-77.15552...   

                                created_at  \
0  {u'$date': u'2017-03-06T11:57:00.000Z'}   
1  {u'$date': u'2017-03-06T11:57:27.000Z'}   
2  {u'$date': u'2017-03-06T11:58:52.000Z'}   
3  {u'$date': u'2017-03-06T11:59:59.000Z'}   
4  {u'$date': u'2017-03-06T12:00:02.000Z'}   

                      created_at_localtime display_text_range  \
0  {u'$date': u'2017-03-06T06:57:00.000Z'}                NaN   
1  {u'$date': u'2017-03-06T06:57:27.000Z'}                NaN   
2  {u'$date': u'2017-03-06T06:58:52.000Z'}                NaN   
3  {u'$date': u'2017-03-06T11:59:59.000Z'}                NaN   
4  {u'$date': u'2017-03-06T07:00:02.000Z'}                NaN   

                                            entities extended_entities  \
0  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
1  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
2  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
3  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
4  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   

  extended_tweet  favorite_count favorited  \
0            NaN               0     False   
1            NaN               0     False   
2            NaN               0     False   
3            NaN               0     False   
4            NaN               0     False   

                         ...                         retweet_count retweeted  \
0                        ...                                     0     False   
1                        ...                                     0     False   
2                        ...                                     0     False   
3                        ...                                     0     False   
4                        ...                                     0     False   

                                              source  \
0  <a href="http://foursquare.com" rel="nofollow"...   
1  <a href="http://instagram.com" rel="nofollow">...   
2  <a href="http://instagram.com" rel="nofollow">...   
3  <a href="http://instagram.com" rel="nofollow">...   
4  <a href="http://foursquare.com" rel="nofollow"...   

                                           sourceFix  \
0  <a href="http://foursquare.com" rel="nofollow"...   
1  <a href="http://instagram.com" rel="nofollow">...   
2  <a href="http://instagram.com" rel="nofollow">...   
3  <a href="http://instagram.com" rel="nofollow">...   
4  <a href="http://foursquare.com" rel="nofollow"...   

                                                text  \
0  Excelente semana de labores a tod@s (@ Institu...   
1  Un dia me dije...ya basta de lo mismo...nunca ...   
2  #PuraRealidad #InicioDeSemana #pensamientodeld...   
3  Just posted a photo @ Playa Del Morro Tumaco N...   
4    I'm at Terpel El Encano https://t.co/RqCF2S7hWc   

                                             textFix   timestamp_ms  \
0  excelente semana de labores a tod@s (@ institu...  1488801420808   
1  un dia me dije...ya basta de lo mismo...nunca ...  1488801447421   
2  #purarealidad #iniciodesemana #pensamientodeld...  1488801532690   
3  just posted a photo @ playa del morro tumaco n...  1488801599394   
4    i'm at terpel el encano https://t.co/rqcf2s7hwc  1488801602744   

                                       timezone_info truncated  \
0  {u'timezone': u'America/Guayaquil', u'obtained...     False   
1  {u'timezone': u'America/Guayaquil', u'obtained...     False   
2  {u'timezone': u'America/Guayaquil', u'obtained...  

In [13]:
df.shape

(4546, 36)

In [14]:
pd.set_option("display.max_columns", 50)
df.head()

_id                                        coordinates  \
0  838720064958443520  {u'type': u'Point', u'coordinates': [-79.20304...   
1  838720176581447681  {u'type': u'Point', u'coordinates': [-79.5333,...   
2  838720534225510404  {u'type': u'Point', u'coordinates': [-78.5833,...   
3  838720814002339840  {u'type': u'Point', u'coordinates': [-78.73340...   
4  838720828053270528  {u'type': u'Point', u'coordinates': [-77.15552...   

                                created_at  \
0  {u'$date': u'2017-03-06T11:57:00.000Z'}   
1  {u'$date': u'2017-03-06T11:57:27.000Z'}   
2  {u'$date': u'2017-03-06T11:58:52.000Z'}   
3  {u'$date': u'2017-03-06T11:59:59.000Z'}   
4  {u'$date': u'2017-03-06T12:00:02.000Z'}   

                      created_at_localtime display_text_range  \
0  {u'$date': u'2017-03-06T06:57:00.000Z'}                NaN   
1  {u'$date': u'2017-03-06T06:57:27.000Z'}                NaN   
2  {u'$date': u'2017-03-06T06:58:52.000Z'}                NaN   
3  {u'$date': u'2017-03-06T11:59:59.000Z'}                NaN   
4  {u'$date': u'2017-03-06T07:00:02.000Z'}                NaN   

                                            entities extended_entities  \
0  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
1  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
2  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
3  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   
4  {u'user_mentions': [], u'symbols': [], u'hasht...               NaN   

  extended_tweet  favorite_count favorited filter_level  \
0            NaN               0     False          low   
1            NaN               0     False          low   
2            NaN               0     False          low   
3            NaN               0     False          low   
4            NaN               0     False          low   

                                                 geo  \
0  {u'type': u'Point', u'coordinates': [-3.994529...   
1  {u'type': u'Point', u'coordinates': [-2.06667,...   
2  {u'type': u'Point', u'coordinates': [-0.25, -7...   
3  {u'type': u'Point', u'coordinates': [1.8292435...   
4  {u'type': u'Point', u'coordinates': [1.160964,...   

                                        id in_reply_to_screen_name  \
0  {u'$numberLong': u'838720064958443520'}                     NaN   
1  {u'$numberLong': u'838720176581447681'}                     NaN   
2  {u'$numberLong': u'838720534225510404'}                     NaN   
3  {u'$numberLong': u'838720814002339840'}                     NaN   
4  {u'$numberLong': u'838720828053270528'}                     NaN   

  in_reply_to_screen_nameFix in_reply_to_status_id in_reply_to_status_id_str  \
0                        NaN                   NaN                       NaN   
1                        NaN                   NaN                       NaN   
2                        NaN                   NaN                       NaN   
3                        NaN                   NaN                       NaN   
4                        NaN                   NaN                       NaN   

  in_reply_to_user_id in_reply_to_user_id_str is_quote_status lang  \
0                 NaN                     NaN           False   es   
1                 NaN                     NaN           False   es   
2                 NaN                     NaN           False  und   
3                 NaN                     NaN           False   es   
4                 NaN                     NaN           False  und   

                                               place possibly_sensitive  \
0  {u'country_code': u'EC', u'url': u'https://api...              False   
1  {u'country_code': u'EC', u'url': u'https://api...              False   
2  {u'country_code': u'EC', u'url': u'https://api...              False   
3  {u'country_code': u'CO', u'url': u'https://api...              False   
4  {u'country_code': u'CO', u'url': u'https://api...      

In [15]:
df["place"].head()

0    {u'country_code': u'EC', u'url': u'https://api...
1    {u'country_code': u'EC', u'url': u'https://api...
2    {u'country_code': u'EC', u'url': u'https://api...
3    {u'country_code': u'CO', u'url': u'https://api...
4    {u'country_code': u'CO', u'url': u'https://api...
Name: place, dtype: object

In [16]:
df["place"].sample()

1253    {u'country_code': u'EC', u'url': u'https://api...
Name: place, dtype: object

In [17]:
place1 = df["place"].sample()
print place1

1712    {u'country_code': u'EC', u'url': u'https://api...
Name: place, dtype: object


In [18]:
place1.get_values()

array([ {u'country_code': u'EC', u'url': u'https://api.twitter.com/1.1/geo/id/009924a469d7ace1.json', u'country': u'Ecuador', u'place_type': u'city', u'bounding_box': {u'type': u'Polygon', u'typeFix': u'polygon', u'coordinates': [[[-80.467605, -3.063716], [-80.467605, -1.962274], [-79.714781, -1.962274], [-79.714781, -3.063716]]]}, u'full_name': u'Guayaquil, Ecuador', u'attributes': {}, u'id': u'009924a469d7ace1', u'name': u'Guayaquil'}], dtype=object)

In [19]:
for value in place1:
    print value
    
len(place1)

{u'country_code': u'EC', u'url': u'https://api.twitter.com/1.1/geo/id/009924a469d7ace1.json', u'country': u'Ecuador', u'place_type': u'city', u'bounding_box': {u'type': u'Polygon', u'typeFix': u'polygon', u'coordinates': [[[-80.467605, -3.063716], [-80.467605, -1.962274], [-79.714781, -1.962274], [-79.714781, -3.063716]]]}, u'full_name': u'Guayaquil, Ecuador', u'attributes': {}, u'id': u'009924a469d7ace1', u'name': u'Guayaquil'}


1

In [20]:
for value in place1:
    print value["name"]

Guayaquil


In [21]:
place1.get_values()[0]["name"]

u'Guayaquil'

In [22]:
place1.get_values()[0]["name"] == "Guayaquil"

True

In [23]:
df["place"].get_values()[0]["name"] == "Guayaquil"

False

In [24]:
df["place"][4]["name"]

u'Pasto'

In [25]:
prueba3 = df["place"].tolist()[:5]
lista_prueba = []
for lugar in prueba3:
    print lugar["name"]
    lista_prueba.append(lugar["name"])

Loja
Milagro
Quito
San Andres de Tumaco
Pasto


In [26]:
lista_prueba

[u'Loja', u'Milagro', u'Quito', u'San Andres de Tumaco', u'Pasto']

In [27]:
print df[df["place"].isnull()].count()
print type(df[df["place"].isnull()]["place"])
print len(df[df["place"].isnull()]["place"])
ciudades = df[~df["place"].isnull()]["place"]
print len(ciudades)
df[df["place"].isnull()]["place"]

_id                           30
coordinates                   30
created_at                    30
created_at_localtime          30
display_text_range             2
entities                      17
extended_entities              0
extended_tweet                 0
favorite_count                30
favorited                     30
filter_level                  30
geo                           30
id                            30
in_reply_to_screen_name        2
in_reply_to_screen_nameFix     1
in_reply_to_status_id          2
in_reply_to_status_id_str      2
in_reply_to_user_id            2
in_reply_to_user_id_str        2
is_quote_status               30
lang                          30
place                          0
possibly_sensitive            15
quoted_status                  0
quoted_status_id               0
quoted_status_id_str           0
retweet_count                 30
retweeted                     30
source                        30
sourceFix                     30
text      

7       NaN
91      NaN
188     NaN
506     NaN
639     NaN
677     NaN
854     NaN
893     NaN
895     NaN
898     NaN
1032    NaN
1087    NaN
1394    NaN
1609    NaN
1651    NaN
1731    NaN
1809    NaN
1855    NaN
2157    NaN
2452    NaN
2709    NaN
2966    NaN
2991    NaN
3087    NaN
3580    NaN
3941    NaN
3987    NaN
4194    NaN
4353    NaN
4487    NaN
Name: place, dtype: object

In [28]:
print type(df[~df["place"].isnull()])
print type(df[~df["place"].isnull()]["place"])

<class 'pandas.core.frame.DataFrame'>
<class 'pandas.core.series.Series'>


In [29]:
df[~df["place"].isnull()]["place"].values

array([ {u'country_code': u'EC', u'url': u'https://api.twitter.com/1.1/geo/id/01da0f365a77a199.json', u'country': u'Ecuador', u'place_type': u'city', u'bounding_box': {u'type': u'Polygon', u'typeFix': u'polygon', u'coordinates': [[[-79.546384, -4.510835], [-79.546384, -3.672156], [-79.100275, -3.672156], [-79.100275, -4.510835]]]}, u'full_name': u'Loja, Ecuador', u'attributes': {}, u'id': u'01da0f365a77a199', u'name': u'Loja'},
       {u'country_code': u'EC', u'url': u'https://api.twitter.com/1.1/geo/id/0039dafe6d412c07.json', u'country': u'Ecuador', u'place_type': u'city', u'bounding_box': {u'type': u'Polygon', u'typeFix': u'polygon', u'coordinates': [[[-79.687083, -2.245476], [-79.687083, -1.96731], [-79.4486, -1.96731], [-79.4486, -2.245476]]]}, u'full_name': u'Milagro, Ecuador', u'attributes': {}, u'id': u'0039dafe6d412c07', u'name': u'Milagro'},
       {u'country_code': u'EC', u'url': u'https://api.twitter.com/1.1/geo/id/0177bc687a9ac697.json', u'country': u'Ecuador', u'place_type

In [30]:
df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto"

0       False
1       False
2       False
3       False
4        True
5       False
6       False
8       False
9       False
10      False
11      False
12      False
13      False
14      False
15      False
16      False
17      False
18      False
19      False
20      False
21      False
22      False
23      False
24      False
25      False
26      False
27      False
28      False
29      False
30      False
        ...  
4516    False
4517    False
4518    False
4519    False
4520    False
4521    False
4522    False
4523    False
4524    False
4525    False
4526    False
4527    False
4528    False
4529    False
4530    False
4531    False
4532    False
4533    False
4534    False
4535    False
4536    False
4537    False
4538    False
4539    False
4540    False
4541    False
4542    False
4543    False
4544    False
4545    False
Name: place, dtype: bool

###### df[df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto"]

    IndexingError: Unalignable boolean Series key provided

In [31]:
print type(df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto")
print type((df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto").values)
(df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto").values

<class 'pandas.core.series.Series'>
<type 'numpy.ndarray'>


array([False, False, False, ..., False, False, False], dtype=bool)

###### df[(df[~df["place"].isnull()]["place"].apply(lambda lugar: lugar["name"]) == "Pasto").values]
    ValueError: Item wrong length 4516 instead of 4546.

In [32]:
df2 = df[~df["place"].isnull()]
print df2.shape

(4516, 36)


In [33]:
for lugar in df2[df2["place"].apply(lambda lugar: lugar["name"]) == "Pasto"]["place"]:
    print lugar["name"]
df2[df2["place"].apply(lambda lugar: lugar["name"]) == "Pasto"].shape

Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto
Pasto


(28, 36)

In [34]:
madrid_tweets = df2[df2["place"].apply(lambda lugar: lugar["name"]) == "Madrid"]
madrid_tweets.shape

(0, 36)

In [35]:
ciudades2 = [lugar["name"] for lugar in ciudades]

In [36]:
len(set(ciudades2))
set(ciudades2)

{u'Alausi',
 u'Aldana',
 u'Alfredo Baquerizo Moreno',
 u'Altamira',
 u'Ambato',
 u'Ancuy\xe1',
 u'Arajuno',
 u'Archidona',
 u'Arenillas',
 u'Atacames',
 u'Azogues',
 u'Babahoyo',
 u'Barbacoas',
 u'Barranca',
 u'Ba\xf1os De Agua Santa',
 u'Bel\xe9n de los Andaqu\xedes',
 u'Bolivar',
 u'Bol\xedvar',
 u'Buena Fe',
 u'Buesaco',
 u'Camilo Ponce Enriquez',
 u'Canoas De Punta Sal',
 u'Carchi',
 u'Cascales',
 u'Castilla',
 u'Cayambe',
 u'Centinela Del Condor',
 u'Chirinos',
 u'Chone',
 u'Chordeleg',
 u'Colombia',
 u'Colta',
 u'Col\xf3n',
 u'Corrales',
 u'Cotacachi',
 u'Crnel. Marcelino Maridue\xf1a',
 u'Cuenca',
 u'Cumanda',
 u'C\xf3rdoba',
 u'Daule',
 u'Duran',
 u'Ecuador',
 u'El Carmen',
 u'El Charco',
 u'El Doncello',
 u'El Guabo',
 u'El Paujil',
 u'El Triunfo',
 u'Empalme',
 u'Esmeraldas',
 u'Florencia',
 u'Garz\xf3n',
 u'Giron',
 u'Gonzanama',
 u'Guaitarilla',
 u'Gualaceo',
 u'Guamote',
 u'Guano',
 u'Guaranda',
 u'Guayaquil',
 u'Guayas',
 u'Huaquillas',
 u'Ibarra',
 u'Ipiales',
 u'Isabela